cloudstack/resource_cloudstack_autoscalevmprofile.go

```diff
@@ -0,0 +1,233 @@
+package cloudstack
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+func resourceCloudStackAutoScaleVMProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudStackAutoScaleVMProfileCreate,
+		Read:   resourceCloudStackAutoScaleVMProfileRead,
+		Update: resourceCloudStackAutoScaleVMProfileUpdate,
+		Delete: resourceCloudStackAutoScaleVMProfileDelete,
+
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"service_offering": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"template": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"other_deploy_params": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"destroy_vm_grace_period": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
Shouldn't this also be computed?

Reply:
Yeah, you're right. I changed it to Computed and there's no need for conditionally setting it on read anymore.